clients/vnetClient/vnetClient.go

```diff
@@ -0,0 +1,49 @@
+package vnetClient
+
+import (
+	"encoding/xml"
+	azure "github.com/MSOpenTech/azure-sdk-for-go"
+)
+
+const (
+	azureNetworkConfigurationURL = "services/networking/media"
+)
+
+//GetVirtualNetworkConfiguration retreives the current virtual network
+//configuration for the currently active subscription. Note that the
+//underlying Azure API means that network related operations are not safe
+//for running concurrently.
+func GetVirtualNetworkConfiguration() (*NetworkConfiguration, error) {
+	response, err := azure.SendAzureGetRequest(azureNetworkConfigurationURL)
+	if err != nil {
+		return nil, err
+	}
+
+	networkConfiguration := &NetworkConfiguration{}
```

Comment:
Could do that - whether constructing from scratch is valid really depends on use case though. Leaving the call in there seems safer though.